boa_tester/src/exec/js262.rs

```diff
@@ -87,11 +89,14 @@ fn eval_script(_this: &JsValue, args: &[JsValue], context: &mut Context) -> JsRe
     // eprintln!("called $262.evalScript()");
 
     if let Some(source_text) = args.get(0).and_then(|val| val.as_string()) {
+        #[cfg(not(feature = "vm"))]
         match boa::parse(source_text.as_str(), false) {
             // TODO: check strict
             Err(e) => context.throw_type_error(format!("Uncaught Syntax Error: {}", e)),
-            Ok(script) => script.run(context),
+            Ok(statement_list) => statement_list.run(context),
         }
+        #[cfg(feature = "vm")]
+        context.eval(source_text.as_str())
```

Comment:
Does this also throw type errors with the "Uncaught Syntax Error: " string? If not, I think it will not replicate the behaviour for parsing errors.

Reply:
You're right, is does not throw a type error. I think I misread this. Nice find, I see how to best solve this best, as we do not have `StatementList::run` for the vm.